### PR TITLE
Add CI for published oauth-server repository

### PR DESCRIPTION
**Pull request by sallyom**
/cc @deads2k @sttts @tnozicka
*  see https://github.com/openshift/origin/pull/23263, will publish to openshift/oauth-server

### CODE REVIEW

**Comment by sallyom** [comment]
we'll be making a clean split from origin, so this PR will be modified to add the oauth-server job/config...

**Comment by sallyom** [comment]
https://github.com/openshift/release/pull/4189#issuecomment-505533696   (applies to this PR also)

File: ci-operator/jobs/openshift/oauth-server/openshift-oauth-server-master-postsubmits.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+postsubmits:
```
Comment (tnozicka) [inline ci-operator/jobs/openshift/oauth-server/openshift-oauth-server-master-postsubmits.yaml]: we should not promote the image while origin CI is promoting the same one, only after moving away from staging

File: ci-operator/config/openshift/oauth-server/openshift-oauth-server-master.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    cluster: https://api.ci.openshift.org
+    name: release
+    namespace: openshift
+    tag: golang-1.12
+canonical_go_repository: github.com/openshift/oauth-server
+images:
+- dockerfile_path: images/Dockerfile.rhel
+  from: base
+  to: oauth-server
+promotion:
```
Comment (tnozicka) [inline ci-operator/config/openshift/oauth-server/openshift-oauth-server-master.yaml]: comment this out

**Comment by tnozicka** [comment]
/hold
for hypershift image to be split in origin as the setting here is dependent on how that will be named, and to be able to override it

File: ci-operator/config/openshift/oauth-server/openshift-oauth-server-master.yaml (hunk 2 above)
Comment (sallyom) [reply ci-operator/config/openshift/oauth-server/openshift-oauth-server-master.yaml]: done

**Comment by sallyom** [comment]
@tnozicka this can be unheld/merged now? I've tested against [this test PR](https://github.com/openshift/oauth-server/pull/1) locally.

**Comment by tnozicka** [comment]
we have the image separated
https://github.com/openshift/release/blob/39892bf/ci-operator/config/openshift/origin/openshift-origin-master.yaml#L39

/approve
/lgtm

(rehearse is chicken-egg)

**Pull request merged**